plutus-ledger/src/Ledger/Constraints/OnChain.hs

```diff
@@ -54,7 +55,7 @@ checkTxConstraint ScriptContext{scriptContextTxInfo} = \case
         $ dv `elem` fmap snd (txInfoData scriptContextTxInfo)
     MustValidateIn interval ->
```

Comment:
We should probably change `MustValidateIn SlotRange` to `MustValidateIn PosixTimeRange`. We won't always be able to compute `slotRangeToPOSIXTimeRange` on-chain without additional information.